.github/workflows/java-gradle-artifact.yaml

```diff
@@ -0,0 +1,171 @@
+name: Reusable workflow for building and publishing java gradle artifacts
+
+on:
+  workflow_call:
+    inputs:
+      java-distribution:
+        description: "Java distribution to be installed. (Default is temurin)"
+        required: false
+        type: string
+        default: "temurin"
+      java-version:
+        description: "Java version to be installed. (Default is 11)"
+        required: false
+        type: string
+        default: "11"
+      gradle-version:
+        description: "Gradle version to be installed. (Default is 6.7.1)"
+        required: false
+        type: string
+        default: "6.7.1"
```

Comment:
We should definitely use gradle wrapper

Reply:
I installed gradle via the action here, because it offers some caching options, but I don't know if these work or have any benefit when using the gradle wrapper. The wrapper is already used everywhere, so the installation here might be pointless

Reply:
By default, the setup-action actually uses `wrapper,` see https://github.com/gradle/gradle-build-action#use-a-specific-gradle-version